snap-jp2-reader/src/main/java/org/esa/snap/jp2/reader/metadata/Jp2XmlMetadata.java

```diff
@@ -134,15 +134,18 @@ public String getProductDescription() {
         return null;
     }
 
+    /**
+     * SNAP-1400 (https://senbox.atlassian.net/browse/SNAP-1400)
+     * SNAP-3459 (https://senbox.atlassian.net/browse/SNAP-3459) - Do not rely on EPSG and check the actual axis name order
+     * Keep in mind that there are JP2 files with values like "  42.788298611111109   23.271622685185189", resulting in empty tokens and tokens starting with white space
+     * @return
+     */
     public Point2D getOrigin() {
         Point2D origin = null;
         String coords = getAttributeValue(JP2ProductReaderConstants.TAG_ORIGIN, null);
         if (coords != null) {
-            // SNAP-1400 (https://senbox.atlassian.net/browse/SNAP-1400)
-            // origin = new Point2D.Double(Double.parseDouble(coords.split(" ")[0]), Double.parseDouble(coords.split(" ")[1]));
-            // there are JP2 files with values like "  42.788298611111109   23.271622685185189", resulting in empty tokens and tokens starting with white space
             final String[] result = Arrays.stream(coords.split(" ")).filter(p -> !StringUtils.isEmpty(p)).map(String::trim).toArray(String[]::new);
-            if (!getCrsGeocoding().equals("EPSG::4326")) {
+            if (!isReversedAxisOrder()) {
```

Comment:
Not using the inversion here saves one operation also one inversion less the brain needs to follow.
Same for the other usages of isReversedAxisOrder()

Reply:
Actually, the intention was to have the general use-case (usual axis order) the first one, and then the reversed axis order use-case.

Reply:
Okay, I understand the intention.
In the end I would still prefer:
```
            String x = !isReversedAxisOrder() ? result[0] : result[1];
            String y = !isReversedAxisOrder() ? result[1] : result[0];
            origin = new Point2D.Double(Double.parseDouble(x), Double.parseDouble(y));
```
over 
```
            if (!isReversedAxisOrder()) {
                // the usual order
                origin = new Point2D.Double(Double.parseDouble(result[0]), Double.parseDouble(result[1]));
            }
            else {
                // reversed order
                origin = new Point2D.Double(Double.parseDouble(result[1]), Double.parseDouble(result[0]));
            }

```
It is less lines of code, and it highlights better what is different between the two cases. But I can accept the current code.

Reply:
Ok, refactoring done :)